components/mediation/tasks/org.wso2.micro.integrator.ntask.core/src/main/java/org/wso2/micro/integrator/ntask/coordination/task/scehduler/CoordinatedTaskScheduler.java

```diff
@@ -89,12 +94,44 @@ public void run() {
                 LOG.debug("This node is not leader. Hence not cleaning task store or resolving un assigned tasks.");
             }
             // schedule all tasks assigned to this node and in state none
-            scheduleAllTasksAssignedToThisNode();
+            scheduleAssignedTasks(CoordinatedTask.States.NONE);
         } catch (Throwable throwable) { // catching throwable to prohibit permanent stopping of the executor service.
             LOG.fatal("Unexpected error occurred while trying to schedule tasks.", throwable);
         }
     }
 
+    /**
+     * Pause ( stop execution ) the deactivated tasks.
+     *
+     * @throws TaskCoordinationException - when something goes wrong while retrieving tasks information from store.
+     */
+    private void pauseDeactivatedTasks() throws TaskCoordinationException {
+
+        List<String> deactivatedTasks = taskStore.retrieveTaskNames(localNodeId, CoordinatedTask.States.DEACTIVATED);
+        if (LOG.isDebugEnabled()) {
+            deactivatedTasks.stream().map(
+                    task -> "Task [" + task + "] retrieved in [" + CoordinatedTask.States.DEACTIVATED + "] state.")
+                    .forEachOrdered(LOG::debug);
+        }
+        ScheduledTaskManager scheduledTaskManager = taskManager;
+        List<String> pausedTasks = new ArrayList<>();
+        List<String> deployedTasks = taskManager.getAllCoordinatedTasksDeployed();
+        deactivatedTasks.forEach(task -> {
+            if (deployedTasks.contains(task)) {
+                try {
+                    scheduledTaskManager.stopExecution(task);
+                    pausedTasks.add(task);
+                } catch (TaskException e) {
+                    LOG.error("Error stopping the task [" + task + "]", e);
+                }
+            } else {
+                LOG.info("The task [" + task + "] retrieved to be paused is not a deployed task "
```

Comment:
same here... Shouldn't we do something about this state?

Reply:
Not really, if it hits this else condition , for some reason the task retrieved from store is not in the deployed list. It could be due to deployment delay or an invalid entry in the store. So we just print info log and ignore , so that task store will clean it out.